src/SFML/Window/Unix/JoystickImpl.cpp

```diff
@@ -554,9 +554,7 @@ JoystickCaps JoystickImpl::getCapabilities() const
     // Get the number of buttons
     char buttonCount = 0;
     ioctl(m_file, JSIOCGBUTTONS, &buttonCount);
-    caps.buttonCount = static_cast<unsigned int>(buttonCount);
-    if (caps.buttonCount > Joystick::ButtonCount)
-        caps.buttonCount = Joystick::ButtonCount;
+    caps.buttonCount = std::min(static_cast<unsigned int>(buttonCount), Joystick::ButtonCount);
```

Comment:
Unrelated change.

Reply:
It's fine. I made this change to fix a clang-tidy warning.

Reply:
Ah I was using a source build of clang-tidy in my Ubuntu 24 machine which meant I had access to checks that don't exist in clang-tidy-18, the version of clang-tidy we use to analyze all the Linux code, hence me seeing a local clang-tidy failure that didn't exist in CI. I reverted this. It's worth fixing but this PR is not the place for that.